iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -300,99 +147,20 @@ private static void InsertEtag(HttpRequestMessage requestMessage, IETagHolder en
             return result;
         }
 
-        private Task PostAsyncHelper<T1>(
+        private async Task PostAsyncHelper<T1>(
             Uri requestUri,
             T1 entity,
             IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> errorMappingOverrides,
```

Comment:
The callers of this method always pass in the same default error mapping for this parameter, so you don't need this to be a parameter anymore